### ISSUE TITLE
doesnt sync/record properly when send mail to another character

### ISSUE THREAD

**Issue by sodade21** [Xruptor/BagSync#130]
when i send something to an alt it doesnt show on tooltip that that alt now has those items on mail. also it doesnt sync well with items on bags when im crafting. shows that i have more at bank while i dont. and it only refresh when i go and open the bank.. is a way around that?

**Comment by Xruptor** [comment]
The issues you are describing are not bugs.  BagSync is working as intended.  I'll do my best to answer your questions though.

In terms of the mail issues, BagSync **does not record outgoing mail**.  This means, it won't track items you send in the mail to a character.  BagSync just tracks newly received items in the mail or items that are left in the Mailbox. This scanning is performed when you open the mailbox.  So if you were to send an item to another character, the item description would only be updated when you login to that character and check your mailbox.

So why is it done this way? Because tracking items in an outgoing setting is quite difficult as there is no real way to determine delivery on the recipients end.  Example, lets say you sent an item to a friend of yours.  How would BagSync know that your friend checked his mail and either grabbed the item from the mailbox or left it there?  If the item was grabbed from your friends Mailbox, then why would the item be listed as Mail in BagSync? When it isn't in the mail anymore?  It's technically not in your possession anymore so why would it categorize it as such?  It gets a little more complicated, but suffice to say, BagSync doesn't track outgoing mail.  I do apologize for any confusion.  I would like to point out that many users keep hundreds of mail in their mailbox because of auctions.  Which they use BagSync to track in the Mailbox.  That is pretty much the primary use of that feature.

In terms of crafting and reagent stuff.  Please see my response in this Issue Ticket.
https://github.com/Xruptor/BagSync/issues/128

In short, there is no way BagSync or any addon can update or scan items from the Reagent window without actually visiting a bank.  I can't make BagSync predict if an item was removed from it because you crafted something, whether the components were from your inventory or partially from your reagents bank.  A lot of things are restricted due to how Blizzard and what they allow addon authors to do.

I hope this clears up any issues you may have.

**Comment by Xruptor** [comment]
v14.9
-Finalized code that will calculate and predict the amount of items that were used while crafting from the Bank and ReagentBank and store it to the database.

-Prediction algorithm should be fairly accurate when consuming items from the Bank and ReagentBank when crafting.  However, it's not 100% foolproof.  It will definitely be 100% accurate once the player visits the bank though.  This is due to limitation imposed by Blizzard on the WOW client.

-Some code cleanup.
